dll/win32/shell32/brfolder.cpp

```diff
@@ -466,18 +536,22 @@ BrFolder_Treeview_Rename(BrFolder *info, NMTVDISPINFOW *pnmtv)
     ASSERT(data);
     ASSERT(BrFolder_GetItemAttributes(info, hItem, SFGAO_CANRENAME) & SFGAO_CANRENAME);
 
+    CComHeapPtr<ITEMIDLIST_ABSOLUTE> pidlOldFull(ILClone(data->pidlFull));
+
     PITEMID_CHILD newChild;
     HRESULT hr = data->lpsfParent->SetNameOf(info->hWnd, data->pidlChild,
                                              pnmtv->item.pszText, SHGDN_NORMAL, &newChild);
     if (FAILED(hr))
         return FALSE;
 
     LPITEMIDLIST newFull;
-    if (SUCCEEDED(hr = SHILClone((LPITEMIDLIST)data->pidlFull, &newFull)))
+    if (SUCCEEDED(hr = SHILClone(data->pidlFull, &newFull)))
     {
         ILRemoveLastID(newFull);
         if (SUCCEEDED(hr = SHILAppend(newChild, &newFull)))
         {
+            SHChangeNotify(SHCNE_RENAMEFOLDER, SHCNF_IDLIST, pidlOldFull, newFull);
```

Comment:
It is `SetNameOf` that should be calling `SHChangeNotify`, not the tree (I know I did this somewhere else as well). We should aim to fully fix notifications everywhere.

Reply:
Fixed.